feat: timer field

zoli opened the pull request:
Timer field type which stores time in minutes. More details about this feature on the related issue #4083.
Resolves #4083.

### Feature Preview

#### PR Checklist

- [x] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [x] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [x] All existing tests are passing.

zoli commented:
https://github.com/AppFlowy-IO/AppFlowy/assets/3286303/19c5d423-9f5b-4d12-8940-8a3f9dd62ea2

Xazin commented on frontend/rust-lib/flowy-database2/src/services/field/type_options/timer_type_option/timer.rs:
```diff
@@ -0,0 +1,171 @@
+    Self
+  }
+
+  pub(crate) fn format_cell_data(minutes: i64) -> String {
```
For this use-case, I'm wondering if we shouldn't do the parsing of the cell in the frontend itself.

Using Intl I believe we can do some pretty nice transformation, including support for different localizations.

See eg. https://github.com/andresaraujo/timeago.dart/blob/master/packages/timeago/lib/src/timeago.dart

zoli replied:
I wasn't sure about this either but doesn't this make it prone to inconsistency between different frontends?

Xazin replied:
You're right, but I think in this case it's the better solution.

In the frontend we have locale information, as well as a better Intl library so we can much easilier parse the time into eg. `5y 100d 2h 31m`. With timestamps we also do parsing on the frontend for dates, we also are able to get timezone information on the frontend quite easily.

In Danish the equivalent would be `5år 100d 2t 31m`.

Xazin commented:
Also something I was thinking about when I saw "Timer", is that I was expecting it to count down. Otherwise I guess calling it "Time" is more relevant.

zoli commented:
> Also something I was thinking about when I saw "Timer", is that I was expecting it to count down. Otherwise I guess calling it "Time" is more relevant.

You are right about timer but I think maybe stopwatch is a better name. I also want to add time tracking feature to this later and it works like count up.

zoli commented:
Thinking about this, It can be a general field type (I'm not sure about the name) which the user can choose to be a timer (count down) or stopwatch (count up) in type options. What do you think @Xazin?

Xazin commented:
> Thinking about this, It can be a general field type (I'm not sure about the name) which the user can choose to be a timer (count down) or stopwatch (count up) in type options. What do you think @Xazin?

Yeah I that is a good idea. Kind of like selecting a number format.

zoli commented:
So I will change the field type name to Time and will add a stopwatch and timer option later in a separate PR.

Xazin commented:
> So I will change the field type name to Time and will add a stopwatch and timer option later in a separate PR.

Sounds good, feel free to ping me when you have renamed the field, I tested and feature works as expected. 👍

I think the time type data can be in minutes, but for Stop timer or Count down timer, it should be a start timestamp.

zoli commented:
@Xazin I renamed the field type to time and moved the formatting function to the flutter frontend.
I will implement these features on other PRs:
- Having precision option (store by minutes or second)
- Timer and Stopwatch as type option
- More localization support as per your review comment.
- Tracking time button for timer and stopwatch
